invenio_records_resources/services/records/service.py

```diff
@@ -288,6 +290,72 @@ def read(self, id_, identity):
             links_tpl=self.links_item_tpl,
         )
 
+    def _read_many(self, identity, es_query, fields=None, record_cls=None,
+                   search_opts=None, **kwargs):
+        """Search for records matching the ids."""
+        # We use create_search() to avoid the overhead of aggregations etc
+        # being added to the query with using search_request().
+        search = self.create_search(
+            identity=identity,
+            record_cls=record_cls or self.record_cls,
+            search_opts=search_opts or self.config.search,
+            permission_action='search',
+        )
+
+        # Fetch only certain fields - explicitly add internal system fields
+        # required to use the result list to dump the output.
+        if fields:
+            dumper_fields = [
+                "uuid", "version_id", "created", "updated", "expires_at"]
+            fields = fields + dumper_fields
+            # ES 7.11+ supports a more efficient way of fetching only certain
+            # fields using the "fields"-option to a query. However, ES 6 and
+            # ES 7 versions does not support it, so we use the source filtering
+            # method instead for now.
+            search = search.source(fields)
+
+        search_result = search.query(es_query).execute()
+
+        return search_result
+
+    def read_many(self, identity, ids, fields=None,
+                  record_cls=None, search_opts=None, **kwargs):
```

Comment:
The high-level service methods ``read_many()`` and ``read_all()`` should not take ``record_cls`` and ``search_opts`` as inputs (it's ok for ``_read_many()`` to do so). Rather, if we need for drafts we would need to add corresponding ``read_all_drafts()`` and ``read_many_drafts()`` methods (however I dont' think we need them, since it's mainly for vocabularies we need them.

Reply:
makes sense, thanks :)